src/dill/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,2161 @@
+                    uint256 timeAfterExpiry = block.timestamp -
+                        thisStake.ending_timestamp;
+
+                    // Get the weighted-average lock_multiplier
+                    uint256 numerator = (lock_multiplier * timeBeforeExpiry) +
+                        (_MultiplierPrecision * timeAfterExpiry);
+                    lock_multiplier =
+                        numerator /
+                        (timeBeforeExpiry + timeAfterExpiry);
+                }
+                // Otherwise, it needs to just be 1x
+                else {
+                    lock_multiplier = _MultiplierPrecision;
+                }
+            } else {
+                uint256 elapsedSeconds = block.timestamp - lastRewardClaimTime;
+                if (elapsedSeconds > 0) {
+                    lock_multiplier = thisStake.isPermanentlyLocked
+                        ? lockMaxMultiplier
+                        : _averageDecayedLockMultiplier(
+                            account,
+                            i,
+                            elapsedSeconds
+                        );
+                    _lastUsedMultiplier[account][i] =
+                        _lastUsedMultiplier[account][i] -
+                        (elapsedSeconds - 1) *
+                        multiplierDecayPerSecond;
+                }
+            }
+            uint256 liquidity = thisStake.liquidity;
+            uint256 combined_boosted_amount = (liquidity * lock_multiplier) /
+                _MultiplierPrecision;
+            lockBoostedDerivedBal =
+                lockBoostedDerivedBal +
+                combined_boosted_amount;
+        }
+
+        return dillBoostedDerivedBal + lockBoostedDerivedBal;
+    }
+
+    /* ========== MUTATIVE FUNCTIONS ========== */
+
+    function kick(address account) public {
+        uint256 _derivedBalance = derivedBalances[account];
+        derivedSupply = derivedSupply - _derivedBalance;
+        _derivedBalance = derivedBalance(account);
+        derivedBalances[account] = _derivedBalance;
+        derivedSupply = derivedSupply + _derivedBalance;
+    }
+
+    function depositAllAndLock(uint256 secs, bool isPermanentlyLocked)
+        external
+        lockable(secs)
+    {
+        _deposit(
+            TOKEN.balanceOf(msg.sender),
+            msg.sender,
+            secs,
+            block.timestamp,
+            isPermanentlyLocked
+        );
+    }
+
+    function depositAll() external {
+        _deposit(
+            TOKEN.balanceOf(msg.sender),
+            msg.sender,
+            0,
+            block.timestamp,
+            false
+        );
+    }
+
+    function depositFor(uint256 amount, address account) external {
+        require(
+            stakingDelegates[account][msg.sender],
+            "Only registered delegates can deposit for their delegator"
+        );
+        _deposit(amount, account, 0, block.timestamp, false);
+    }
+
+    function depositForAndLock(
```

Comment:
What are some scenarios where a delegated deposit is useful? Since the user holding the tokens will have to approve the gauge contract for spending the token, they might as well go all the way and ultimately perform the deposit themselves right, rather than waiting for someone to eventually do it?

Reply:
Needs discussion. Although these functions cause no harm and might support an use case of infinite approval to the platform and the platform auto renewing your rewards as deposits.